File: PulseSocial-Frontend/src/services/bookmarkService.js
import api from "../api/axios";

export const toggleBookmark = (postId, userId, collectionName = "Favorites") => {
    const query = collectionName ? `?collectionName=${encodeURIComponent(collectionName)}` : "";
    return api.post(`/bookmarks/${postId}${query}`);
};

export const getBookmarks = (userId) => {
    return api.get(`/bookmarks/user/${userId}`);
};

export const getCollections = (userId) => {
    return api.get(`/bookmarks/collections/${userId}`);
};

File: src/main/java/com/pulsesocial/service/BookmarkService.java
package com.pulsesocial.service;

import java.util.List;
import com.pulsesocial.dto.PostResponse;

public interface BookmarkService {
    String toggleBookmark(Long userId, Long postId, String collectionName);
    List<PostResponse> getBookmarks(Long userId);
    List<String> getCollections(Long userId);
}
